lib/utils/slice.go

```diff
@@ -142,3 +142,12 @@ func FromSlice[T any](r []T, key func(T) string) map[string]T {
 
 	return out
 }
+
+// MapKeysToSlice extracts keys of a map as a slice.
```

Comment:
`x/exp/maps` already has a `maps.Keys` implementation that does this, and we are already using it in other parts of Teleport.

Note that it did not make the standard library version of `maps`..